File: backend-Gestion_Post/models/Post.js
const mongoose = require('mongoose');

const postSchema = new mongoose.Schema({
    
        title:{
            type: String,
            required: true
        },
        desc:{
            type: String,
            required:true
        },
        image:{
            type: String,
            required:false
        },
        category:{
            type: String,
            required:false
        },
        date: {
            type: String, // Consider using type: Date if you want to store actual date objects
            default: () => Date.now(), // Use a function to ensure it gets the current date at runtime
            required: true
        },
        state:{
            type: String,
            required:true
        }
    
        
    },
    {
        timestamps:true
    });

const postModel = mongoose.model('Post', postSchema);

module.exports = postModel;

File: GestionPost/app/src/main/java/tn/esprit/gestionpost/Data/Post.kt
package tn.esprit.gestionpost.Data

enum class PostState {
    FRIENDS_ONLY,
    PRIVATE,
    PUBLIC
}

data class Post(
    val _id: String,
    val title: String,
    val desc: String,
    val image: String,
    val category: String?,
    val date: String,
    val state: PostState


) {
    override fun toString(): String {
        return "Post(_id='$_id', title='$title', desc='$desc', image='$image', category=$category, date='$date', state=$state)"
    }
}